src/user/views/adsManager/views/advanced/components/review/components/AdSetReview.tsx

```diff
@@ -1,21 +1,45 @@
-import { AdSetForm, Creative, OS, Segment } from "../../../../../types";
-import { FormikErrors } from "formik";
+import {
+  AdSetForm,
+  CampaignForm,
+  Creative,
+  OS,
+  Segment,
+} from "../../../../../types";
+import { FormikErrors, useField, useFormikContext } from "formik";
 import { ConversionDisplay } from "components/Conversion/ConversionDisplay";
 import { ReviewField } from "./ReviewField";
 import { ReviewContainer } from "user/views/adsManager/views/advanced/components/review/components/ReviewContainer";
+import { CampaignFormat } from "graphql/types";
+import { CreativeSpecificPreview } from "components/Creatives/CreativeSpecificPreview";
+import { useEffect } from "react";
 
 interface Props {
   idx: number;
   adSet: AdSetForm;
+  format: CampaignFormat;
   errors?: string | FormikErrors<AdSetForm>;
 }
 
 export function AdSetReview({ adSet, idx, errors }: Props) {
+  const { setFieldValue } = useFormikContext<CampaignForm>();
+  const [, creatives] = useField<Creative[]>("creatives");
+  const hasErrors = !!errors;
+  useEffect(() => {
+    adSet.creatives.forEach((c, index) => {
+      const found = creatives.value.find((f) => f.id === c.id);
+      if (found) {
+        void setFieldValue(`adSets.${idx}.creatives.${index}`, {
+          ...found,
+          creativeInstanceId: c.creativeInstanceId,
+        });
+      }
+    });
+  }, []);
```

Comment:
I'm uncomfortable that I'd understand what this code is trying to achieve in the future, and so it's likely to break. It feels like a sticking plaster for something else not quite being right?